ui/app/adapters/dynamic-host-volume.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import WatchableNamespaceIDs from './watchable-namespace-ids';
+import classic from 'ember-classic-decorator';
+
+@classic
+export default class DynamicHostVolumeAdapter extends WatchableNamespaceIDs {
```

Comment:
this generally matches what we have for our CSI adapter. We could have used the existing volume adapter and added some "if it has an ID / if it's a dynamic host volume" conditions, but figured it was better to give it its own file. Also true of model, serializer.